### PR TITLE
feat(nuget): look for refs within package.props schema

### PR DESCRIPTION
**Pull request by edwinf**
WIth the [Microsoft.Build.CentralPackageVersions](https://github.com/microsoft/MSBuildSdks/tree/master/src/CentralPackageVersions) SDK, there are additional properties for nuget references.  This update will honor GlobalPackageReference and PackageReference where the schema calls for an Update and not an Include for the package reference in a package.props file.

A live test was performed against this repository: [edwinf/live-test-renovate-packages.props](https://github.com/edwinf/live-test-renovate-packages.props)

### CODE REVIEW

File: lib/manager/nuget/extract.ts [hunk 1]
```diff
@@ -74,7 +74,7 @@ async function determineRegistryUrls(
   return registryUrls;
 }
 
-const packageRe = /<(?:PackageReference|DotNetCliToolReference).*Include\s*=\s*"([^"]+)".*Version\s*=\s*"(?:[[])?(?:([^"(,[\]]+)\s*(?:,\s*[)\]]|])?)"/;
+const packageRe = /<(?:PackageReference|DotNetCliToolReference|GlobalPackageReference).*(?:Include|Update)\s*=\s*"([^"]+)".*Version\s*=\s*"(?:[[])?(?:([^"(,[\]]+)\s*(?:,\s*[)\]]|])?)"/;
```
Comment (viceice) [inline lib/manager/nuget/extract.ts:77]: This regex should be refactored to use regex groups for better readability and easy extension

**Review by viceice — CHANGES_REQUESTED** [review]

**Comment by viceice** [comment]
We should support `VersionOverride` too.

File: lib/manager/nuget/extract.ts (hunk 1 above)
Comment (edwinf) [reply lib/manager/nuget/extract.ts:77]: I'll admit, regex isn't my strong suit so I'm not quite sure what you mean, it's already using the match groups further down in the code based on this regex.  Is there an example somewhere else in the code of what you'd like that refactor to look like?

Comment (viceice) [reply lib/manager/nuget/extract.ts]: ok, i'll update the regex later

File: lib/manager/nuget/extract.ts [hunk 2]
```diff
@@ -74,7 +74,7 @@ async function determineRegistryUrls(
   return registryUrls;
 }
 
-const packageRe = /<(?:PackageReference|DotNetCliToolReference).*Include\s*=\s*"([^"]+)".*Version\s*=\s*"(?:[[])?(?:([^"(,[\]]+)\s*(?:,\s*[)\]]|])?)"/;
+const packageRe = /<(?:PackageReference|DotNetCliToolReference|GlobalPackageReference).*(?:Include|Update)\s*=\s*"([^"]+)".*(?:Version|VersionOverride)\s*=\s*"(?:[[])?(?:([^"(,[\]]+)\s*(?:,\s*[)\]]|])?)"/;
```
Comment (viceice) [inline lib/manager/nuget/extract.ts:77]:
```suggestion
const packageRe = /<(?:PackageReference|DotNetCliToolReference|GlobalPackageReference).*(?:Include|Update)\s*=\s*"(?<depName>[^"]+)".*(?:Version|VersionOverride)\s*=\s*"(?:[[])?(?:(?<currentValue>[^"(,[\]]+)\s*(?:,\s*[)\]]|])?)"/;
```

Now you can change line 141-142 to
```ts
      const { currentValue, depName } = match.groups;
```

**Review by viceice — CHANGES_REQUESTED** [review]
The regex name group suggestion

File: lib/manager/nuget/extract.ts (hunk 2 above)
Comment (edwinf) [reply lib/manager/nuget/extract.ts]: Changes applied!

**Review by viceice — APPROVED** [review]

**Pull request merged**